### ISSUE TITLE
Update software dependencies

### ISSUE THREAD

**Issue by julie-sullivan** [opencb/cellbase#413]
dependency | version in use | update to 
-- | -- | -- 
[Mongodb](https://www.mongodb.com/support-policy) |  3.4.20 | 4.2
Neo4J |  | 3.5
JUnit | 4.12 | 5.5
Java | 7 | 8
[Maven](https://maven.apache.org/docs/history.html) |  | 3.6
Tomcat | 7 | 9
Checkstyle | 6.1.1 | 8.19

Also to be discussed:

 * Jackson
 * SLF
 * jersey

Should these be individual tickets?

**Comment by imedina** [comment]
I do not think we need different issues for tracking this. Maybe we could just use a task list

**Comment by imedina** [comment]
Note that Jackson, SLF and MongoDB Java Driver versions are taken from java-commons-lib, we should not redefine them if possible.
We always try to use the same base dependency version across OpenCB. For instance, the MongoDB Java driver (not MongoDB database) is now https://github.com/opencb/java-common-libs/blob/develop/commons-datastore/commons-datastore-mongodb/pom.xml#L26

**Comment by julie-sullivan** [comment]
I am using the latest mongo (4.2) and maven (3.6.2) without any issues. I'm also using tomcat 8.5 but haven't tested the web services locally yet.
